subsys/net/l2/ethernet/gptp/gptp_md.c

```diff
@@ -583,6 +583,12 @@ static void gptp_md_pdelay_req_state_machine(int port)
 		state->state = GPTP_PDELAY_REQ_NOT_ENABLED;
 	}
 
+	while (1) {
+		if (net_if_flag_is_set(GPTP_PORT_IFACE(port), NET_IF_UP)) {
+			break;
+		}
+		k_sleep(K_MSEC(1000));
+	}
```

Comment:
I am not sure this is a good idea, this will block the calling thread. Why can't we just return if the interface is not up?

Reply:
@hakehuang , can you provide your insights on this?

Reply:
I have just pushed an alternative proposal for that. I basically removed that while/sleep combo, and replaced it with this:

https://github.com/zephyrproject-rtos/zephyr/pull/42216/commits/44ea9cd76a54cfd3b0fecbcd4c32fbc900c8160e

Does it look good?